app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,610 @@
+            }
+
+            setLoading(false);
+        }, General.SEARCH_TIMEOUT_MILLISECONDS);
+    }, [dataSource]);
+
+    const getMultiselectData = useCallback((multiselectSelectedElems: MultiselectSelectedMap): Selection => {
+        let myItems;
+        let multiselectItems: Selection = [];
+
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS:
+                myItems = multiselectSelectedElems as Dictionary<UserProfile>;
+                multiselectItems = multiselectItems as UserProfile[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            case ViewConstants.DATA_SOURCE_CHANNELS:
+                myItems = multiselectSelectedElems as Dictionary<Channel>;
+                multiselectItems = multiselectItems as Channel[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            default:
+                myItems = multiselectSelectedElems as Dictionary<DialogOption>;
+                multiselectItems = multiselectItems as DialogOption[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+        }
+    }, [multiselectSelected, dataSource]);
+
+    // Effects
+    useNavButtonPressed(SUBMIT_BUTTON_ID, componentId, onHandleMultiselectSubmit, [onHandleMultiselectSubmit]);
+
+    useEffect(() => {
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            getChannels();
+        } else if (!loading && !term) {
+            searchDynamicOptions('');
+        }
+    }, []);
+
+    useEffect(() => {
+        let listData: (DataType | UserProfileSection[]) = integrationData;
+
+        if (term) {
+            listData = searchResults;
+        }
+
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            listData = createProfilesSections(listData as UserProfile[]);
+        }
+
+        if (!dataSource || dataSource === ViewConstants.DATA_SOURCE_DYNAMIC) {
+            listData = (integrationData as DialogOption[]).filter((option) => option.text && option.text.toLowerCase().includes(term));
+        }
+
+        setCustomListData(listData);
+    }, [searchResults, integrationData]);
+
+    useEffect(() => {
+        if (!isMultiselect) {
+            return;
+        }
+
+        setButtons(componentId, {
+            rightButtons: [rightButton],
+        });
+    }, [rightButton, componentId, isMultiselect]);
+
+    useEffect(() => {
```

Comment:
What exactly is doing this effect? At first glance it feels it can easily get into an infinite loop.